brainiak/fcma/classifier.py

```diff
@@ -0,0 +1,278 @@
+                                                      num_voxels,
+                                                      num_voxels,
+                                                      num_TRs,
+                                                      1.0, data,
+                                                      num_voxels, num_voxels,
+                                                      0.0, corr_data,
+                                                      num_voxels, count)
+            count += 1
+        logger.debug(
+            'correlation computation done'
+        )
+        # normalize if necessary
+        if self.epochs_per_subj > 0:
+            corr_data = corr_data.reshape(1,
+                                          num_samples,
+                                          num_voxels * num_voxels)
+            fcma_extension.normalization(corr_data, self.epochs_per_subj)
+            corr_data = corr_data.reshape(num_samples, num_voxels, num_voxels)
+            logger.debug(
+                'normalization done'
+            )
+        # training
+        if isinstance(self.clf, sklearn.svm.SVC) \
+                and self.clf.kernel == 'precomputed':
+            kernel_matrix = np.zeros((num_samples, num_samples),
+                                     np.float32,
+                                     order='C')
+            # for using kernel matrix computation from voxel selection
+            corr_data = corr_data.reshape(1,
+                                          num_samples,
+                                          num_voxels * num_voxels)
+            blas.compute_kernel_matrix('L', 'T',
+                                       num_samples, num_voxels * num_voxels,
+                                       1.0, corr_data,
+                                       0, num_voxels * num_voxels,
+                                       0.0, kernel_matrix, num_samples)
+            data = kernel_matrix
+            # training data is in shape [num_samples, num_voxels * num_voxels]
+            self.training_data = corr_data.reshape(num_samples,
+                                                   num_voxels * num_voxels)
+            logger.debug(
+                'kernel computation done'
+            )
+        else:
+            data = corr_data.reshape(num_samples, num_voxels * num_voxels)
+        self.clf = self.clf.fit(data, y)
+        time2 = time.time()
+        logger.info(
+            'training done, takes %.2f s' %
+            (time2 - time1)
+        )
+        return self
+
+    def predict(self, X):
+        """ use a trained model to predict correlation data
+
+        the input data X is activity data filtered by top voxels
+        and prepared for correlation computation.
+        X needs to be first converted to correlation,
+        and then normalized across all samples in the list
+        if len(X) > 1,
+        and then predicted via self.clf.
```

Comment:
If this is the description of what the method does, change to phrasing from "needs to be" to "is".